File: backend/app/models/treatment_plan.py
from pydantic import BaseModel, Field
from typing import List, Optional

class TreatmentRecommendation(BaseModel):
    name: str = Field(..., description="Name of the exercise or treatment")
    description: str = Field(..., description="Detailed description of the exercise or treatment")
    sets: str = Field(..., description="Number of sets to perform")
    reps: str = Field(..., description="Number of repetitions or duration per set")
    frequency: str = Field(..., description="How often to perform the exercise (e.g., Daily, Twice daily, Weekly)")
    duration: str = Field(..., description="How long to continue this exercise (e.g., 2 weeks, Ongoing)")
    precautions: str = Field(..., description="Safety precautions and warning signs to watch for")

class TreatmentPlan(BaseModel):
    treatment_focus: str = Field(
        ...,
        description="Primary focus of the treatment plan (e.g., pain management, strength building, flexibility)"
    )
    treatment_recommendations: List[TreatmentRecommendation] = Field(
        ...,
        description="List of specific exercises and treatments recommended"
    )
    reasoning: str = Field(
        ...,
        description="Explanation of why this treatment approach was chosen"
    )
    next_phase_focus: str = Field(
        ...,
        description="Focus for the next phase of treatment after completing this plan"
    )

    class Config:
        json_schema_extra = {
            "example": {
                "treatment_focus": "pain management and core stability",
                "treatment_recommendations": [
                    {
                        "name": "Pelvic Tilts",
                        "description": "Gentle core exercise to improve pelvic stability and reduce lower back strain",
                        "sets": "3",
                        "reps": "10-12",
                        "frequency": "Daily",
                        "duration": "2 weeks",
                        "precautions": "Stop if pain increases beyond 3/10. Maintain neutral spine position."
                    },
                    {
                        "name": "Cat-Cow Stretch",
                        "description": "Dynamic spinal mobility exercise to improve flexibility and reduce stiffness",
                        "sets": "2",
                        "reps": "Hold each position for 5-10 seconds",
                        "frequency": "Twice daily",
                        "duration": "2 weeks",
                        "precautions": "Move slowly and breathe deeply. Avoid if sharp pain occurs."
                    },
                    {
                        "name": "Bridging Exercise",
                        "description": "Progressive core and glute strengthening exercise",
                        "sets": "3",
                        "reps": "Hold for 10 seconds",
                        "frequency": "Daily",
                        "duration": "Ongoing",
                        "precautions": "Maintain neutral spine. Stop if pain increases in lower back."
                    }
                ],
                "reasoning": "This initial treatment plan focuses on pain management and core stability to address the reported lower back pain. The exercises are chosen to be gentle yet effective, with a focus on proper form and gradual progression. The plan includes both mobility and stability exercises to address both immediate pain relief and long-term strength building.",
                "next_phase_focus": "progressive strengthening and functional movement patterns"
            }
        } 

File: frontend/app/treatment-plan/components/TreatmentPlan.js
import React from 'react';

const TreatmentPlan = ({ treatmentPlan }) => {
  const { treatment_focus, treatment_recommendations, reasoning, next_phase_focus } = treatmentPlan;

  return (
    <div className="bg-card shadow-sm border border-slate-200 rounded-lg p-6 space-y-6">
      {/* Treatment Focus */}
      <div className="border-b pb-4">
        <h2 className="text-xl font-semibold text-card-foreground mb-2">Current Focus</h2>
        <p className="text-foreground capitalize">{treatment_focus}</p>
      </div>

      {/* Exercises */}
      <div className="space-y-6">
        <h2 className="text-xl font-semibold text-card-foreground">Recommended Exercises</h2>
        <div className="space-y-4">
          {treatment_recommendations.map((exercise, index) => (
            <div key={index} className="bg-muted border border-slate-100 p-4 rounded-lg hover:shadow-md transition-shadow cursor-pointer">
              <h3 className="text-lg font-medium text-card-foreground mb-2">{exercise.name}</h3>
              <div className="space-y-2">
                <p className="text-muted-foreground"><span className="font-medium">Description:</span> {exercise.description}</p>
                <p className="text-muted-foreground"><span className="font-medium">Sets:</span> {exercise.sets}</p>
                <p className="text-muted-foreground"><span className="font-medium">Reps:</span> {exercise.reps}</p>
                <p className="text-muted-foreground"><span className="font-medium">Frequency:</span> {exercise.frequency}</p>
                <p className="text-muted-foreground"><span className="font-medium">Duration:</span> {exercise.duration}</p>
                <p className="text-muted-foreground"><span className="font-medium">Precautions:</span> {exercise.precautions}</p>
              </div>
            </div>
          ))}
        </div>
      </div>

      {/* Clinical Reasoning */}
      <div className="border-t pt-4">
        <h2 className="text-xl font-semibold text-card-foreground mb-2">Clinical Reasoning</h2>
        <p className="text-muted-foreground">{reasoning}</p>
      </div>

      {/* Next Phase */}
      <div className="border-t pt-4">
        <h2 className="text-xl font-semibold text-card-foreground mb-2">Next Phase Focus</h2>
        <p className="text-foreground capitalize">{next_phase_focus}</p>
      </div>
    </div>
  );
};

export default TreatmentPlan; 